src/Raven.Studio/typescript/viewmodels/database/status/ongoingTasksStats.ts

```diff
@@ -127,6 +166,42 @@ class hitTest {
         this.rTree.insert(data);
     }
 
+    registerSubscriptionExceptionItem(x: number, y: number, width: number, height: number, exceptionItem: subscriptionExceptionItemInfo) { 
+        const data = {
+            minX: x,
+            minY: y,
+            maxX: x + width,
+            maxY: y + height,
+            actionType: "subscriptionExceptionItem",
+            arg: exceptionItem
+        } as rTreeLeaf;
+        this.rTree.insert(data);
+    }
+
+    registerSubscriptionPendingItem(x: number, y: number, width: number, height: number, pendingItem: subscriptionPendingItemInfo) {
```

Comment:
I think it is time to introduce extra method for registering rTreeLeaf to avoid doing:
```
const data = {
            minX: x,
            minY: y,
            maxX: x + width,
            maxY: y + height,
            actionType: XXXXXX
            arg: item
        } as rTreeLeaf;
        this.rTree.insert(data);
```

so many times.

Reply:
done